infra/src/main/kotlin/fi/fta/geoviite/infra/split/SplitService.kt

```diff
@@ -637,24 +648,30 @@ private fun calculateTopologicalConnectivity(
 
 private fun findSplitIndices(
     alignment: LayoutAlignment,
-    startSwitch: Pair<IntId<TrackLayoutSwitch>, JointNumber>?,
-    endSwitch: Pair<IntId<TrackLayoutSwitch>, JointNumber>?,
+    startSwitch: SplitPointSwitch?,
+    endSwitch: SplitPointSwitch?,
 ): IntRange {
     val startIndex = startSwitch?.let { (s, j) -> findIndex(alignment, s, j) } ?: 0
     val endIndex = endSwitch?.let { (s, j) -> findIndex(alignment, s, j) - 1 } ?: alignment.segments.lastIndex
 
-    return if (startIndex < 0 || endIndex > alignment.segments.lastIndex || endIndex < startIndex) {
```

Comment:
Itse "korjaus" on tässä. Ei siis estä käyttäjää saamasta 400-virhettä, mutta viesti on sentään vähän kuvaavampi. Kyseisessä bugicasessa siis jakokohdaksi valittua vaihdetta ei voida uudelleenlinkittää koska geometria on niin hapan. Vaihde on pakko korjata käsin ennen splittausta ja on kyllä jo tehtävälistalla tarjolla.

Reply:
Näkyyköhän tuo uudelleenlinkityksen epäonnistuminen sillä tavalla, että ihan jo vaihde-ehdotuksen generoiminen epäonnistuisi? Tein toisella tiketillä sen, että sellaiset tilanteet joissa joillekin vaihteille ei saada luotua vaihde-ehdotusta automaattilinkityksessä, käsitellään splittauksen kannalta kriittisinä virheitä ja tällöin disabloidaan koko splittaus-UI tehtävälistan avaamislinkkiä lukuunottamatta. Eli hyvässä tapauksessa tälle caselle saattaa nyt olla aiempaa parempi UX.

Reply:
En ole ihan varma, mutta epäilen että ei. Minusta näytti siltä että se saa tuolle vaihde-ehdotuksen, koska tätä kohtaa aiemmin tehdään jo checkit että kaikki vaihteet linkittyy jonnekin. Linkitys on vain sikäli jojossa että tämä jaettava raide ei ole sen jälkeen enää kiinni tuossa vaihteessa -> sen mukaan ei voida jakaa. Johtunee siitä että raiteet ovat erillään toisistaan ja vaihteen esityspiste on vanhastaan sen toisen raiteen alussa.